File: C++/PalindromeNumber.cc
// Determine whether an integer is a palindrome. Do this without extra space.

class Solution {
public:
    bool isPalindrome(int x) {
        int t = x / 10;
        int d = 1;
        while (t) {
            t /= 10;
            d *= 10;
        }
        while (d > 1) {
            if ((x / d) % 10 != x % 10) return false;
            x /= 10;
            d /= 100;
        }
        return true;
    }
};

File: Python/palindrome_number.py
#!/usr/bin/env python

# Determine whether an integer is a palindrome. Do this without extra space.

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        return x >= 0 and int(str(x)[::-1]) == x
